### PR TITLE
ZIP32 test vectors

### PR DESCRIPTION
**Pull request by Milan-Cerovsky**
# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), [Architecture.md](../blob/main/docs/Architecture.md), etc.)
- [ ] **Run the demo app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [README.md](../blob/main/README.md), [Architecture.md](/blob/main/docs/Architecture.md), etc. as appropriate
- [ ] **Run the demo app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the SDK, some aspects require manual testing. If you had to manually test 
something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the demo app to look for build failures or crashes, humans running the demo app are 
more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt [hunk 1]
```diff
@@ -26,19 +27,45 @@ class DerivationToolImplTest {
...
+            assertEquals("1xm/qCXZqXgiRFT1IVk97+5gv9BE5AjkOVWHwYU9RYQ=", Base64.encode(key),)
+        }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    @Test
+    fun testVectorDerivedArbitraryWalletKey() =
+        runTest {
+            val secretKey =
+                DerivationTool.getInstance().deriveArbitraryWalletKey(
+                    contextString = "5a63617368207465737420766563746f7273".hexToByteArray(),
```
Comment (str4d) [inline sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt:51]: For at least one of the two test vectors, can you use `"Zcash test vectors"` converted into a byte array? That's how Zashi Android is constructing its context string, and I want to ensure we test that this conversion works the same way in Kotlin and Rust.

(or both, and test with both this and hex)

**Review by str4d — CHANGES_REQUESTED** [review]

File: sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt (hunk 1 above)
Comment (str4d) [reply sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt:51]: This is what I mean:
```suggestion
                    contextString = "Zcash test vectors".toByteArray(),
```

Comment (str4d) [reply sdk-incubator-lib/src/androidTest/kotlin/cash/z/ecc/android/sdk/internal/DerivationToolImplTest.kt:51]: I want the above change specifically to ensure that this code in Zashi Android is not breaking its specification by silently encoding the string in an unexpected byte format: https://github.com/Electric-Coin-Company/zashi-android/pull/1643/files#diff-055de5dc228c0e87d964a8ec4865d3e48d0c20067cc3e6e03061d096cf481ec5R69

**Review by str4d — APPROVED** [review]
utACK b2d3d3701feba3be22d47319123a633a04de11cc

**Review by HonzaR — APPROVED** [review]

**Pull request merged**